product_pricelist_supplierinfo/tests/test_product_supplierinfo.py

```diff
@@ -0,0 +1,332 @@
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Product Test",
+                "seller_ids": [
+                    (0, 0, {"partner_id": cls.supplier1.id, "min_qty": 5, "price": 50}),
+                    (0, 0, {"partner_id": cls.supplier2.id, "min_qty": 1, "price": 10}),
+                ],
+            }
+        )
+        cls.product_with_diff_uom = cls.env["product.product"].create(
+            {
+                "name": "Product UOM Test",
+                "uom_id": cls.env.ref("uom.product_uom_unit").id,
+                "uom_po_id": cls.env.ref("uom.product_uom_dozen").id,
+                "seller_ids": [
+                    (
+                        0,
+                        0,
+                        {"partner_id": cls.supplier1.id, "min_qty": 1, "price": 1200},
+                    ),
+                ],
+            }
+        )
+        cls.pricelist = cls.env["product.pricelist"].create(
+            {
+                "name": "Supplierinfo Pricelist",
+                "item_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "compute_price": "formula",
+                            "base": "supplierinfo",
+                            "price_discount": 0,
+                            "min_quantity": 1.0,
+                        },
+                    ),
+                ],
+            }
+        )
+
+    @classmethod
+    def _update_rate(cls, currency_id, rate):
+        currency_rate = cls.currency_rate_obj.search(
+            [("name", "=", date.today()), ("currency_id", "=", currency_id.id)], limit=1
+        )
+        if not currency_rate:
+            cls.currency_rate_obj.create(
+                {"currency_id": currency_id.id, "rate": rate, "name": date.today()}
+            )
+        else:
+            currency_rate.write({"rate": rate})
+
+    def test_pricelist_based_on_product_category(self):
+        self.pricelist.item_ids[0].write(
+            {
+                "price_discount": 50,
+                "applied_on": "2_product_category",
+                "categ_id": self.env.ref("product.product_category_all").id,
+            }
+        )
+        self.assertAlmostEqual(
+            self.pricelist._get_product_price(self.product, 1),
+            5.0,
+        )
+
+    def test_pricelist_based_on_product(self):
+        self.pricelist.item_ids[0].write(
+            {
+                "applied_on": "1_product",
+                "product_tmpl_id": self.product.product_tmpl_id.id,
+            }
+        )
+        self.assertAlmostEqual(
+            self.pricelist._get_product_price(self.product, 1),
+            10.0,
+        )
+        self.assertAlmostEqual(
+            self.product.product_tmpl_id.with_context(
+                pricelist=self.pricelist.id
+            )._get_contextual_price(),
+            10.0,
```

Comment:
Remove this assertion, this is not required anymore. It was to check value was propagated to the pricelist field and, since such field doesn't exist anymore, there's no point on checking it.

Same in all cases below.

It would be a good idea to document this in commit description

Reply:
I see you documented this in commit description, you just need to split lines to 72 characters.

On the other hand, I'd suggest something more clear, e.g.
```
Assertion in tests involving the pricelist field in products were
removed, asthat field doesn't exist anymore [1], and such assertions
were intended only to ensure computed price were propagated correctly to
that field.

[1] https://github.com/odoo/odoo/commit/9e99a9df
```